blogService/src/main/java/com/blog/blogService/service/impl/ArticleServiceImpl.java

```diff
@@ -0,0 +1,32 @@
+package com.blog.blogService.service.impl;
+
+import com.blog.blogService.dto.ArticleDto;
+import com.blog.blogService.entity.Article;
+import com.blog.blogService.entity.Blog;
+import com.blog.blogService.mapper.ArticleMapper;
+import com.blog.blogService.repository.ArticleRepository;
+import com.blog.blogService.repository.BlogRepository;
+import com.blog.blogService.service.spi.ArticleService;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@AllArgsConstructor
+@Service
+public class ArticleServiceImpl implements ArticleService {
+
+    private final ArticleRepository articleRepository;
+    private final ArticleMapper articleMapper;
+    private final BlogRepository blogRepository;
+
+    @Override
+    public Long createArticle(final ArticleDto articleDto, final Long blogId) {
+        Article articleToSave = articleMapper.map(articleDto, Article.class);
+        Optional<Blog> optionalBlog = blogRepository.findById(blogId);
+        if (optionalBlog.get().isPrivateBlog() == false)
```

Comment:
proszę dodawaj {} nawet jeśli blok ma jedną linijkę - poprawia to czytelność i w przyszłości jeśli byś musiała dodać coś do tego bloku będzie to łatwiejsze

Reply:
ponad to - wywołujesz get() bez wcześniejszego isPresent co na pewnoe powoduje warning w IntelliJ, proponuje dołączyć to do tej linijki funkcyjnej, w taki sposób:
```
        blogOptional.ifPresent(blog -> {
            if(!blog.isPrivat()) {
                articleToSave.setArticleBlog(blog);
            }
        });
```

Reply:
dodatkowo - jeśli masz do czynienia ze zmienną boolean - unijak porównywania jej wartosci do true/false. sama ta zmienna ma już taką wartość więc może posłużyć w warunku - tak jak napisałam powyżej - IF wykona się tylko wtedy gdy wartość isPrivate jest false, a ponieważ stosujemy negację, warunek zostaje ewaluowany do true więc wykona się jego blok :)

Reply:
Oczywiscie :)
Zrobione